velox/exec/tests/RowContainerTest.cpp

```diff
@@ -508,12 +555,83 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     std::vector<std::optional<T>> ascNullsFirstOrder = {
         std::nullopt, lowest, 0.0, min, max, inf, nan};
 
-    testOrderAndNullsFirstVariations<T, T>(
+    testOrderAndEqualsWithNullsFirstVariations<T>(
         type, values, ascNullsFirstOrder, [&](const auto& expectedOrder) {
           return makeNullableFlatVector<T>(expectedOrder);
         });
   }
 
+  template <typename T>
+  void populateFloatingPointTestVectors(
+      int32_t numRows,
+      std::vector<std::optional<T>>& rawData,
+      std::vector<bool>& rawExpected) {
+    TEST_FLOATING_TYPE_LIMIT_VARIABLES;
```

Comment:
This macro results in a lot of warnings of unusued variables. Can you replace it with something like : auto [max, min, _ , _,_ , _] = FLOATING_TYPE_LIMIT_VARIABLES();  
Where the function returns a const std::tuple and you only give names to the ones you need ?

Reply:
I tried to use the tuple approach. The issue is that if you use more than one `_` it complains because the identifier repeats. It doesn't exclude it. And from posts I read there should be warnings for `_` as well as it is simply just a name for one of the returned entries.

Example:
```
/Users/czentgr/gitspace/velox/velox/exec/tests/RowContainerTest.cpp:630:25: error: redefinition of '_'
    const auto [nan, _, _, _, _] = getTestFloatingTypeLimitVariables<T>();
                        ^
```

I turned on`-Wunused-variables` to see what the compiler will do and on macOS (clang) it doesn't trigger the error if the entries were named but not used. However, the linter might still complain.

In most functions all the special values should be used except in a few where only NaN is used. So I will just not use the macro there. I experimented with a templated struct instead of the macro but this also causes more changes.